minpy/primitive.py

```diff
@@ -319,3 +374,60 @@ def gradable(self, bp_args, bp_kwargs):
         """
         return all(i in self._grad_func for i in bp_args) and \
                all(i in self._grad_func_kw for i in bp_kwargs)
+
+
+def customop(op_type):
+    """Wrapper for CustomOp.
+
+    Parameters
+    ----------
+    type : str
+        Type of the custom op. Input vars are converted to the given type. 'minpy' or 'mxnet'.
+
+    Returns
+    -------
+    Returns the proper wrapper by given type.
+    """
+
+    def customop_wrapper(func):
+        """Real wrapper"""
+
+        class ArgWrappedPrimitive(Primitive):
+            """Wrapped func in __call__ method."""
+
+            def __call__(self, *args, **kwargs):
+                return super(ArgWrappedPrimitive, self).__call__(args, kwargs)
+
+        if op_type == 'numpy' or op_type == 'mxnet':
+            return ArgWrappedPrimitive(func, variants[op_type], custom_op=True)
+        else:
+            raise ValueError('Wrong type for CustomOp. Must be either "numpy" or "mxnet"')
+
+    return customop_wrapper
```

Comment:
It won't work. The `__call__` signature must be changed from `Primitive`, or the user need to supply `args` and `kwargs` to there funcs.